plaidml/bridge/openvino/src/plaidml_plugin/plaidml_util.cpp

```diff
@@ -0,0 +1,234 @@
+  const auto p = desc.getPrecision();
+
+#define CASE(prec) \
+  case prec:       \
+    return InferenceEngine::make_shared_blob<PrecisionTrait<prec>::value_type>(desc);
+
+  switch (p) {
+    CASE(Precision::FP32);
+    CASE(Precision::FP16);
+    CASE(Precision::Q78);
+    CASE(Precision::U16);
+    CASE(Precision::U8);
+    CASE(Precision::I8);
+    CASE(Precision::BOOL);
+    CASE(Precision::I32);
+    CASE(Precision::I64);
+    CASE(Precision::BIN);
+
+    default:
+      THROW_IE_EXCEPTION << "The plugin does not support input " << p.name() << " precision";
+  }
+}
+
+/*
+
+
+plaidml::exec::Binding util::make_binding(const std::string& dev, const TensorDesc& desc) {
+  auto ph = Placeholder(util::to_plaidml(desc));
+
+  // LogicalShape shape(ffi::call<plaidml_logical_shape*>(plaidml_logical_shape_clone, arg.shape));
+  // TensorShape tensor_shape(shape.dtype(), shape.sizes());
+
+  auto buff = plaidml::Buffer(dev, {ph.shape().dtype(), ph.shape().sizes()});
+
+  return plaidml::exec::Binding{std::move(ph), std::move(buff)};
+
+  /*Tensor tensor(ffi::call<plaidml_expr*>(plaidml_expr_clone, arg.tensor));
+      LogicalShape shape(ffi::call<plaidml_logical_shape*>(plaidml_logical_shape_clone, arg.shape));
+      ProgramArgument programArg{arg.is_input, tensor, shape, nullptr};
+      if (arg.buffer) {
+        TensorShape tensor_shape(shape.dtype(), shape.sizes());
+        auto bufptr = ffi::call<plaidml_buffer*>(plaidml_buffer_clone, arg.buffer);
+        programArg.buffer = std::make_shared<Buffer>(bufptr, tensor_shape);
+      }
+
+
+* /
+
+    /*
+    LogicalShape(DType dtype, const std::vector<int64_t>& dims)
+          : ptr_(details::make_ptr(ffi::call<plaidml_logical_shape*>(
+                plaidml_logical_shape_alloc, static_cast<plaidml_datatype>(dtype), dims.size(), dims.data()))) {}
+    */
+
+plaidml::edsl::LogicalShape util::to_plaidml(const TensorDesc& desc) {
+  return {static_cast<plaidml::DType>(util::to_plaidml(desc.getPrecision())), util::to_plaidml(desc.getDims())};
+}
+
+std::vector<int64_t> util::to_plaidml(const SizeVector& dims) { return std::vector<int64_t>{dims.begin(), dims.end()}; }
+
+std::vector<int32_t> util::to_plaidml(const PropertyVector<unsigned int>& property) {
+  std::vector<int32_t> property_vec;
+  for (int i = 0; i < property.size(); ++i) {
+    property_vec.push_back(property[i]);
+  }
+  return property_vec;
+}
+
+util::OpType util::str_to_type(const std::string& layer_type) {
+  std::unordered_map<std::string, util::OpType, details::CaselessHash<std::string>, details::CaselessEq<std::string>>
+      type_table = {
+          {"convolution", util::OpType::Convolution},
+          {"pooling", util::OpType::Pooling},
+          {"scaleshift", util::OpType::ScaleShift},
+          {"relu", util::OpType::ReLU},
+          {"eltwise", util::OpType::Eltwise},
+          {"reshape", util::OpType::Reshape},
+          {"innerproduct", util::OpType::Fc},
+          {"fullyconnected", util::OpType::Fc},
```

Comment:
Unless OpenVINO considers "innerproduct" and "fullyconnected" exactly synonymous, we should probably have these map to two different implementations that call shared core code after minor cleanup, verification, etc.